crates/gpui/src/platform/linux/platform.rs

```diff
@@ -282,23 +285,23 @@ impl<P: LinuxClient + 'static> Platform for P {
                     .directory(options.directories)
                     .send()
                     .await
-                    .ok()
-                    .and_then(|request| request.response().ok())
-                    .and_then(|response| {
-                        response
-                            .uris()
-                            .iter()
-                            .map(|uri| uri.to_file_path().ok())
-                            .collect()
-                    });
-
+                    .map(|request| {
+                        request.response().ok().and_then(|response| {
+                            response
+                                .uris()
+                                .iter()
+                                .map(|uri| uri.to_file_path().ok())
+                                .collect()
+                        })
+                    })
+                    .map_err(|_| anyhow!(FILE_PICKER_PORTAL_MISSING));
```

Comment:
This is actually not correct, we would need https://github.com/bilelmoussaoui/ashpd/issues/209 first. As with your current approach, cancelling the dialog by the user (closing it) would show the same error message as well

Reply:
Made a new release that covers this & a backport of your last patch, thanks!

Reply:
Thanks a lot! I didn't notice originally since I had also coded other part of the logic incorrectly and the cancellation error didn't show up, it should be fixed now.